framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -188,7 +188,27 @@ private static String formatAnnotations(Collection<? extends Annotation> annos)
      */
     private static String formatArrayType(ATypeElement e, String arrayType) {
         StringBuilder result = new StringBuilder();
-        return formatArrayTypeImpl(e, arrayType, result);
+        int componentEndPos;
+        if (arrayType.startsWith("@")) {
+            // if there's an explicit annotation, go to second space
+            componentEndPos = arrayType.indexOf(' ', arrayType.indexOf(' ') + 1);
+        } else {
+            // otherwise, go to first space
+            componentEndPos = arrayType.indexOf(' ');
+        }
+
+        // if the first array doesn't have an annotation, then the spacing will be off,
```

Comment:
What is the first array?  Do you mean the first array level?

(Nit: please capitalize sentences to make them easier to read.)